sympy/functions/elementary/complexes.py

```diff
@@ -140,12 +166,30 @@ class im(Function):
     >>> from sympy.abc import x, y
     >>> im(2*E)
     0
-    >>> re(2*I + 17)
-    17
+    >>> im(2*I + 17)
+    2
     >>> im(x*I)
     re(x)
     >>> im(re(x) + y)
     im(y)
+    >>> im(x + y*I)
+    re(y) + im(x)
+    >>> im(2 + 3*I).as_real_imag()
+    (3, 0)
+    >>> im(21*I + 50).as_real_imag(deep=True)
+    (21, 0)
+
+    Parameters
+    ==========
+
+    exp : expression
```

Comment:
```suggestion
    arg : Expr
```
I have seen that if there is doc is available for `Expr` somewhere, it gets collected by some frameworks like napoleon, so we should pinpoint to actual classes (like `Expr`) instead of using notations like (`expression`)

I'd also suggest that the parameter name should match the actual variable name used in `__new__` or (`eval` for sympy functions)

Reply:
Yes.

Reply:
Done.